src/worker.cpp

```diff
@@ -1563,4 +1563,103 @@ void Worker::performBackupAndRestore(types::VPDMapVariant& io_srcVpdMap)
         */
     }
 }
+
+void Worker::deleteFruVpd(const std::string& i_dbusObjPath)
+{
+    if (i_dbusObjPath.empty())
+    {
+        throw std::runtime_error("Given DBus object path is empty.");
+    }
+
+    const std::string& l_fruPath =
+        jsonUtility::getFruPathFromJson(m_parsedJson, i_dbusObjPath);
+
+    try
+    {
+        const bool l_presentPropValue =
+            std::get<bool>(dbusUtility::readDbusProperty(
+                constants::pimServiceName, i_dbusObjPath,
+                constants::inventoryItemInf, "Present"));
+
+        if (!l_presentPropValue)
+        {
+            logging::logMessage(
+                "Given FRU is not present. Could not perform delete FRU VPD for " +
+                i_dbusObjPath);
+            return;
+        }
+        else
+        {
+            if (jsonUtility::isActionRequired(m_parsedJson, l_fruPath,
+                                              "preAction", "deletion"))
+            {
+                if (!jsonUtility::processSystemCmdTag(m_parsedJson, l_fruPath,
```

Comment:
Why this API is being called directly? Why not executeBaseAction?

Reply:
Updated